ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright 2023 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+
+(function() {
+    "use strict";
+
+    let dataLayer;
+    let formName, fieldName, fieldType, panelName;
+    let startTime = (new Date()).getTime();
+
+    const FASTTRACK_ANALYTICS_EVENT = "FormFastTrackAnalyticsEvent";
+    const FormEvents = {
+        RENDER: "Form renditions",
+        SUBMIT: "Form submissions",
+        ERROR: "Validation errors",
+        FIELD: "Field visits",
+        ABANDON: "Abandoned forms",
+        HELP: "Help views",
+        TIMESPENT: "Time Spent",
+        SAVE: 'Drafts' // TODO: implement when portal is available for CC
+    }
+
+    const getFormName = () => {
+        const guidePath = guideBridge.getFormModel().getState().properties['fd:path'].substring(0,guideBridge.getFormModel().getState().properties['fd:path'].indexOf('/jcr:content/guideContainer'));
```

Comment:
what is the definition of form name?
 there is no such thing as name in the schema.